backend/server.js

```diff
@@ -693,6 +704,60 @@ async function findMatchingUser(requestLoginKey) {
   return null;
 }
 
+// sends data to the database in the backend
+app.post('/sendUserData', wrap(async (req, res) => {
+  // Don't cache this endpoint.
+  res.setHeader('Cache-Control', 'no-cache, no-store');
+
+
+  // if there's no body in the request, well, we'll crash, so let's not
+  if (!req.body || !req.body.loginKey) {
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return;
+  }
+
+  // Checks checks checks
+  // Make sure the login key is valid
+  if (typeof req.body.loginKey !== 'string' || req.body.loginKey.length !== 100) {
```

Comment:
We need to make sure that the given user loginKey actually matches one of the keys that are in the DB, too ;)